samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -143,4 +145,43 @@ public boolean exists(String path) {
   public void close() throws ZkInterruptedException {
     zkClient.close();
   }
+
+  /**
+    * subscribe for changes of JobModel version
+    * @param dataListener describe this
+    */
+  public void subscribeToJobModelVersionChange(IZkDataListener dataListener) {
+    LOG.info("pid=" + processorId + " subscribing for jm version change at:" + keyBuilder.getJobModelVersionPath());
+    zkClient.subscribeDataChanges(keyBuilder.getJobModelVersionPath(), dataListener);
+  }
+
+  /**
+   * read the jobmodel version from ZK
+   * @return jobmodel version as a string
+   */
+  public String getJobModelVersion() {
+    return zkClient.<String>readData(keyBuilder.getJobModelVersionPath());
+  }
+
+  /**
+   * verify that given paths exist in ZK
+   * @param paths
+   */
+  public void makeSurePersistentPathsExists(String[] paths) {
+    for (String path : paths) {
+      if (!zkClient.exists(path)) {
+        zkClient.createPersistent(path, true);
```

Comment:
actually if we pass second parameter 'true' then:
"..if true all parent dirs are created as well and no {@link ZkNodeExistsException} is thrown in case the  path already exists"